libraries/AP_HAL_Linux/RCInput_Aero.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+/* feature added by Anemos Technologies : Philippe Crochat (pcrochat@anemos-technologies.com) */
+
+#include "RCInput_Aero.h"
+#include <stdio.h>
+
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+using namespace Linux;
+
+#define DEBUG 1
+
+#if DEBUG
+#define debug(fmt, args...) ::printf(fmt "\n", ##args)
+#else
+#define debug(fmt, args...)
+#endif
+
+static const AP_HAL::HAL &hal = AP_HAL::get_HAL();
+
+
+RCInput_Aero::RCInput_Aero()
+{
+	
+}
+
+void RCInput_Aero::init()
+{
+	const uint8_t tx_buffer[2]={CLOCK_REG, 0x00};
+	uint8_t rx_buffer[2];
+	
+	_spi = hal.spi->get_device(DEVICE_NAME);
+	
+	if (!_spi) {
+        AP_HAL::panic("Could not initialize Aero RCInput");
+    }
+	
+    if (!_spi->transfer_fullduplex(tx_buffer, rx_buffer, 2)) {
+    	AP_HAL::panic("Could not get FPGA frequency");
+    }
+    
+    _frequency=(uint8_t)rx_buffer[1];
+    
+    if (!_frequency) {
+    	AP_HAL::panic("Aero RCInput : Panic no valid FPGA frequency read for device name");
+    }
+}
+
+void RCInput_Aero::_timer_tick(void)
```

Comment:
I don't think doing the spi transaction on the timer tick would be ideal. Replace this with a _spi->register_periodic_task() in the init like in other devices.

Reply:
ok

Reply:
Because they aren't on a spi/i2c bus.

Reply:
ok

Reply:
After looking at all the different RCInputs, all use the _timer_tick function even the very slow RCInput_UART. Does that make really sense to use a different method for the SPI?

Reply:
it's better, but both would work.